MRN-Code/coinstac
# Subscriptions

* **Problem:**
   1. Users will want to see changes to certain data sets - consortia, pipelines, etc. - as they happen.
* **Solution:**
   1. Create RethinkDB changefeeds on certain tables at server start.
   1. Use Apollo Server subscription to listen for those changes and publish messages to listening clients
   1. Use Apollo Client to listen for subscription changes - both for general changes and specific document changes within those tables.
   1. Add functionality to delete pipelines and include subs for pipeline deletion.
   1. Add functionality to delete pipeline step covariate inputs and include subs to signal their deletion.
   1. Remove some unused code from previous prototype and organize/refactor dashboard and user-related components into their respective directories.
   1. Added some additional permission checks to server side delete
   1. Refactor and consolidate shared client-side delete and save functions.
   1. Modals added on deletion for consortia, collections, pipelines
* **Tests:**
   1. Seed new data from `coinstac-api-server` via: `npm run test-setup`
   1. Start server: `npm run start`
   1. Start two instances of the GUI - A & B.
   1. Login on both machines with user: `author` and pass: `password`
   1. Go to the Consortia tab on both instances and on instance **A** click `Create Consortia`.
   1. Fill out `name` and `description` and hit save on instance **A**. Notice the new Consortia appearing in the list on instance **B**.
   1. View details on the newly created consortium on instance **B**.
   1.Make a change to the name on instance **B** and hit save. See the change appear on instance **A**.
   1. Repeat process for Pipelines and Computations. (Computation will be slightly different as the UI differs).
